Custom pod labels

philipsd6 opened the pull request:
This should resolve #563. If/when this passes muster, I'll squash so it can be merged.

philipsd6 committed:
Enable custom labels to be applied to the pods

* Add pod_labels to CRD
* Add pod_labels and comment to defaults
* Add pod_labels to deployment.yaml template
* Add section for pod_labels to README

shanemcd commented on roles/installer/defaults/main.yml:
```diff
@@ -64,6 +64,11 @@ hostname: ''
 
+# Custom labels for the AWX pods. Specify as literal block. E.g.:
+# pod_labels: |
+#   prometheus.io/scape: 'true'
```
Did you mean `scrape`?

philipsd6 replied:
Yes, sure did -- fixed, and PR updated.

shanemcd commented:
Hi, sorry for the slow review. Looks like this has conflicts now. Once you fix, we'll get this reviewed and merged.

philipsd6 commented:
> Hi, sorry for the slow review. Looks like this has conflicts now. Once you fix, we'll get this reviewed and merged.

No problem @shanemcd -- I've gone ahead and rebased on devel without conflicts as requested, and squashed down to one commit.

tchellomello reviewed (approved):
Very nice @philipsd6!! Thanks for your contribution!!

jsalatiel commented:
Any chance we can get this merged ?

timansky commented:
I am having the same problem. #1160 must resolve issue. waiting for merge

fosterseth commented:
additional labels was implemented in this PR https://github.com/ansible/awx-operator/pull/1160

does that PR address the use case you had in mind @philipsd6

philipsd6 commented:
@fosterseth I believe it does, as PR #1160 enables setting `additional_labels` which as I understand it would then apply (from the example in my README) update `prometheus.io/scrape-awx` to the pods, which was our use-case for this PR. However, the `additional_labels` solution will also apply it to the service as well, but I can't see why that would be a problem, so I would consider our use-case met by PR #1160.
